examples/synch-subscription.cpp

```diff
@@ -100,14 +140,24 @@ int main(int argc, char **argv) {
         std::cout << "subscription confirmed\n";
         rx_buff.consume(parse_result.consumed);
 
+        // endless receive loop
         while (true) {
+            // blocks until new message
             parse_result = c.read(rx_buff);
+            // extract (aka decouple from buffer / copy) result
             auto extract = boost::apply_visitor(r::extractor<Iterator>(),
                                                 parse_result.result);
+            // safe to consume buffer now
             rx_buff.consume(parse_result.consumed);
+
+            // select the result in the form of
+            // pair<const std::string&, const std::string&>
             auto payload = boost::apply_visitor(payload_extractor(), extract);
             if (payload) {
-                std::cout << "received payload:: '" << *payload << "'\n";
+                // and finally print it
+                std::cout << "on channel '" << payload->first
+                          << "' received payload :: '" << payload->second
+                          << "'\n";
```

Comment:
Urghhhh. The example outputs something like 

```
subscribed
subscription confirmed
on channel 'channel-one' received payload :: '{"a": 5555}'
```

> Also, I suggest boost::string_ref or std::string_view instead of std::string const&

In this example (`synch-subscription.cpp`) it operates on *extracted* results, i.e. results already unbounded from buffer, so, they are **already**  `std::string`, so, return `const std::string&` seems natural here. 

In the example near by (`stream-parse.cpp`) int operates on yet **bounded** to buffer results So, there you'll find something like `boost::string_ref`. But, as the rx_buff might be scattered / fragmented, it cannot return just `boost::string_ref` as the last requires continuous/flattened memory region. 

It is possible, however, apply some visitor, to let it flattens `parsed_result`s, and then returns `boost::string_ref`s, if needed; although I don't see rationale in that as `r::extractor<Iterator>()` does mostly the same, but with **ownership semantics** (i.e. the result of extraction will be `std::string`).